src/GenericConnectionHandler.cpp

```diff
@@ -46,6 +46,12 @@ bool GenericConnectionHandler::updateSetting(const models::NetworkSetting& s) {
     }
 }
 
+void GenericConnectionHandler::getSetting(models::NetworkSetting& s) {
+    if(_ch != nullptr) {
+        _ch->getSetting(s);
+    }
```

Comment:
In case `_ch` is null I would set type of s to be `NetworkAdapter::NONE`, so that we can understand that the internal pointer is not instantiated yet.

Reply:
Done